include/seqan3/alphabet/aminoacid/translation.hpp

```diff
@@ -0,0 +1,187 @@
+constexpr aa27 translate_triplet(nucl_type const & n1, nucl_type const & n2, nucl_type const & n3) noexcept
+{
+    return seqan3::detail::translation_table<nucl_type, gc>::VALUE[n1.to_rank()][n2.to_rank()][n3.to_rank()];
+}
+
+/*!\brief Translate one nucleotide triplet into single amino acid (tuple interface)
+ * \tparam tuple_type The type of input tuple
+ * \param input_tuple Triplet of nucleotides that should be converted to amino acid
+ * \details
+ *
+ * Translates std::tuple or std::array with 3 nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * No-throw guarantee.
+*/
+template <typename tuple_type, genetic_code gc = genetic_code::CANONICAL>
+    requires std::tuple_size<tuple_type>::value == 3 &&
+             nucleotide_concept<std::tuple_element_t<0, tuple_type>> &&
+             nucleotide_concept<std::tuple_element_t<1, tuple_type>> &&
+             nucleotide_concept<std::tuple_element_t<2, tuple_type>>
+constexpr aa27 translate_triplet(tuple_type const & input_tuple) noexcept
+{
+    return translate_triplet(std::get<0>(input_tuple), std::get<1>(input_tuple), std::get<2>(input_tuple));
+}
+
+/*!\brief Translate one nucleotide triplet into single amino acid (range interface)
+ * \tparam range_type The type of input range
+ * \param input_range Range of three nucleotides that should be converted to amino acid
+ *
+ * \details
+ *
+ * Translates range with 3 nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * Strong exception guarantee (never modifies data).
+*/
+template <typename range_type, genetic_code gc = genetic_code::CANONICAL>
+    requires input_range_concept<range_type> &&
+             nucleotide_concept<std::decay_t<ranges::range_reference_t<std::decay_t<range_type>>>>
+constexpr aa27 translate_triplet(range_type && input_range)
+{
+    assert(ranges::size(input_range) == 3);
+    auto n1 = ranges::begin(input_range);
+    auto n2 = ++n1;
+    auto n3 = ++n2;
+    return translate_triplet(*n1, *n2, *n3);
+}
+
+
+/*!\brief Translate one nucleotide triplet into single amino acid (range interface, input range allows random access)
+ * \tparam range_type The type of input range
+ * \param input_range Range of three nucleotides that should be converted to amino acid
+ *
+ * \details
+ *
+ * Translates range with 3 nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * Strong exception guarantee (never modifies data).
+*/
+template <typename range_type, genetic_code gc = genetic_code::CANONICAL>
+    requires random_access_range_concept<range_type> &&
+             nucleotide_concept<std::decay_t<ranges::range_reference_t<std::decay_t<range_type>>>>
+constexpr aa27 translate_triplet(range_type && input_range)
+{
+    assert(ranges::size(input_range) == 3);
```

Comment:
same thing here (yes, ranges can be random access and not have size o_O).
instead assert that begin, begin+1, begin+2 are all != end